File: clients/Java-GraalVM-native/src/main/java/model/EntityAction.java
package model;

import util.StreamUtil;

public class EntityAction {
    private model.MoveAction moveAction;
    public model.MoveAction getMoveAction() { return moveAction; }
    public void setMoveAction(model.MoveAction moveAction) { this.moveAction = moveAction; }
    private model.BuildAction buildAction;
    public model.BuildAction getBuildAction() { return buildAction; }
    public void setBuildAction(model.BuildAction buildAction) { this.buildAction = buildAction; }
    private model.AttackAction attackAction;
    public model.AttackAction getAttackAction() { return attackAction; }
    public void setAttackAction(model.AttackAction attackAction) { this.attackAction = attackAction; }
    private model.RepairAction repairAction;
    public model.RepairAction getRepairAction() { return repairAction; }
    public void setRepairAction(model.RepairAction repairAction) { this.repairAction = repairAction; }
    public EntityAction() {}
    public EntityAction(model.MoveAction moveAction, model.BuildAction buildAction, model.AttackAction attackAction, model.RepairAction repairAction) {
        this.moveAction = moveAction;
        this.buildAction = buildAction;
        this.attackAction = attackAction;
        this.repairAction = repairAction;
    }
    public static EntityAction readFrom(java.io.InputStream stream) throws java.io.IOException {
        EntityAction result = new EntityAction();
        if (StreamUtil.readBoolean(stream)) {
            result.moveAction = model.MoveAction.readFrom(stream);
        } else {
            result.moveAction = null;
        }
        if (StreamUtil.readBoolean(stream)) {
            result.buildAction = model.BuildAction.readFrom(stream);
        } else {
            result.buildAction = null;
        }
        if (StreamUtil.readBoolean(stream)) {
            result.attackAction = model.AttackAction.readFrom(stream);
        } else {
            result.attackAction = null;
        }
        if (StreamUtil.readBoolean(stream)) {
            result.repairAction = model.RepairAction.readFrom(stream);
        } else {
            result.repairAction = null;
        }
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        if (moveAction == null) {
            StreamUtil.writeBoolean(stream, false);
        } else {
            StreamUtil.writeBoolean(stream, true);
            moveAction.writeTo(stream);
        }
        if (buildAction == null) {
            StreamUtil.writeBoolean(stream, false);
        } else {
            StreamUtil.writeBoolean(stream, true);
            buildAction.writeTo(stream);
        }
        if (attackAction == null) {
            StreamUtil.writeBoolean(stream, false);
        } else {
            StreamUtil.writeBoolean(stream, true);
            attackAction.writeTo(stream);
        }
        if (repairAction == null) {
            StreamUtil.writeBoolean(stream, false);
        } else {
            StreamUtil.writeBoolean(stream, true);
            repairAction.writeTo(stream);
        }
    }
}

File: clients/Kotlin-GraalVM-native/src/main/kotlin/model/EntityAction.kt
package model

import util.StreamUtil

class EntityAction {
    var moveAction: model.MoveAction? = null
    var buildAction: model.BuildAction? = null
    var attackAction: model.AttackAction? = null
    var repairAction: model.RepairAction? = null
    constructor() {}
    constructor(moveAction: model.MoveAction?, buildAction: model.BuildAction?, attackAction: model.AttackAction?, repairAction: model.RepairAction?) {
        this.moveAction = moveAction
        this.buildAction = buildAction
        this.attackAction = attackAction
        this.repairAction = repairAction
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): EntityAction {
            val result = EntityAction()
            if (StreamUtil.readBoolean(stream)) {
                result.moveAction = model.MoveAction.readFrom(stream)
            } else {
                result.moveAction = null
            }
            if (StreamUtil.readBoolean(stream)) {
                result.buildAction = model.BuildAction.readFrom(stream)
            } else {
                result.buildAction = null
            }
            if (StreamUtil.readBoolean(stream)) {
                result.attackAction = model.AttackAction.readFrom(stream)
            } else {
                result.attackAction = null
            }
            if (StreamUtil.readBoolean(stream)) {
                result.repairAction = model.RepairAction.readFrom(stream)
            } else {
                result.repairAction = null
            }
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        val moveAction = moveAction;
        if (moveAction == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            moveAction.writeTo(stream)
        }
        val buildAction = buildAction;
        if (buildAction == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            buildAction.writeTo(stream)
        }
        val attackAction = attackAction;
        if (attackAction == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            attackAction.writeTo(stream)
        }
        val repairAction = repairAction;
        if (repairAction == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            repairAction.writeTo(stream)
        }
    }
}
